File: leetcode_js/MoveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let anchor = 0;
    for (let traverse = 0; traverse < nums.length; traverse++) {
        if (nums[traverse] != 0) {
            let temp = nums[traverse];
            nums[traverse] = nums[anchor];
            nums[anchor] = temp;

            anchor++;
        }
    }
};

// Tests
console.log(moveZeroes([0,1,0,3,12]));

File: leetcode_py/MoveZeroes.py
from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        anchor = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                temp = nums[i]
                nums[i] = nums[anchor]
                nums[anchor] = temp
                anchor += 1
        return nums

# Tests
example = Solution()
print(example.moveZeroes([1,3,9,0,7,0,6]))
